src/control/cmd/daos/container.go

```diff
@@ -708,6 +718,10 @@ func newContainerInfo(poolUUID, contUUID *uuid.UUID) *containerInfo {
 	ci.RedundancyFactor = (*uint32)(&ci.dci.ci_redun_fac)
 	ci.NumSnapshots = (*uint32)(&ci.dci.ci_nsnapshots)
 	ci.HighestAggregatedEpoch = (*uint64)(&ci.dci.ci_hae)
+	ci.dci.ci_snapshots = C.makeUint64Array(4096)
```

Comment:
As noted earlier, if you use the cgo wrappers for memory allocation, then there is less defensive programming required. I believe this should work equivalently:
```go
	ci.dci.ci_snapshots = (*C.uint64_t)(C.calloc(4096, C.sizeof_uint64_t))
```

Reply:
Also, I recommend adding a new method on `containerInfo` to handle cleaning things up rather than relying on users of `containerInfo` to know/remember all of the details. Users still need to remember to call the cleanup method, but at least that boils it down to a single detail.